backend/endpoints/design_assistant.py

```
@@ -0,0 +1,724 @@
+            "message": "BOM data ingested successfully",
+            "summary": {
+                "total_weight_lb": processed_data["weight_metrics"]["total_weight"],
+                "parts_analyzed": processed_data["weight_metrics"]["rows_counted"],
+                "missing_material_count": len(processed_data["missing_material_parts"]),
+            },
+        }
+
+    except Exception as e:
+        if isinstance(e, (ClientException, ServerException)):
+            raise e
+        raise ServerException(f"Failed to ingest BOM data: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)